android/src/main/java/io/customer/reactnative/sdk/RNCIOInAppMessaging.kt

```diff
@@ -0,0 +1,78 @@
+package io.customer.reactnative.sdk
+
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import io.customer.messaginginapp.type.InAppEventListener
+import io.customer.messaginginapp.type.InAppMessage
+
+/**
+ * ReactNative module to hold in-app messages features in a single place to bridge with native code.
+ */
+class RNCIOInAppMessaging(
+    private val reactContext: ReactApplicationContext,
+) : ReactContextBaseJavaModule(reactContext), InAppEventListener {
+    private var listenerCount = 0
```

Comment:
I only see `listenerCount` being used to ignore a `sendEvent` call if there are no listeners attached. 

Since this code base doesn't have automated tests written for it, I prefer we avoid logic such as a Int we manually increment and decrement. We could easily introduce a bug into the code in the future that breaks the event listener feature. 

Would it hurt to remove this `listenerCount` and it's usage in this class?

Reply:
The `listenerCount` is suggested by [react native docs](https://reactnative.dev/docs/native-modules-android#sending-events-to-javascript). I believe the count can mainly be used for cases when we can register/unregister from actual source. Since, we can only register to source on initialization, I added skipping sending JS events based on it. As we don't modify the count directly, we only update it based on what value we receive from react native calls. While the value for removing listener is passed by react native itself, we never call the remove method directly.

Reply:
Although I do see that `listenerCount` is inside of the RN docs, I am still wondering: Would it hurt to remove this `listenerCount` and it's usage in this class?

Reply:
I think it is still helpful as for customers not adding any listener for in-app, the count will help us prevent emitting events unnecessarily. So I'm keeping this unless we found it breaking something 🤞🏻